db/migrate/20220505070823_add_confirmed_at_to_transfer_packages.rb

```diff
@@ -0,0 +1,21 @@
+class AddConfirmedAtToTransferPackages < ActiveRecord::Migration
+  def up
+    add_column :transfer_packages, :confirmed_at, :datetime, null: true
+    add_index :transfer_packages, :confirmed_at
+
+    TransferPackage.reset_column_information
+
+    # Fill confirmed_at on existing packages if their sample transfers have been confirmed
+    TransferPackage.find_each do |package|
+      if confirmed_at = package.sample_transfers.map(&:confirmed_at).compact.max
+        package.update_columns(
+          confirmed_at: confirmed_at,
+        )
+      end
```

Comment:
That looks really good :+1:
But I'm not sure it's worth it. Would need to validate the migration again. And I don't think there is much practical relevance. AFAIK this will land in production before loads of transfer packages are used in practice, right?

Reply:
Yes, as we both said it's not worth it, I just couldn't shut up the optimizer in my brain! I wrote a bunch of such data migrations in the past, and it was the difference between dozens of minutes down to some seconds.